packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1027,7 +1028,8 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
   @mustCallSuper
   FocusAttachment attach(BuildContext? context, {FocusOnKeyCallback? onKey}) {
     _context = context;
-    this.onKey = onKey ?? this.onKey;
```

Comment:
I also thought about it for a long time before I realized the problem, haha

Reply:
@gspencergoog Hi, I wanna ask what should we do if both `Focus.onKey` and `FocusNode.onKey` are non-null?

Reply:
Event the `Focus.onKey` is null?

Reply:
@gspencergoog Hi,
The processing of the first build and subsequent update processing is inconsistent. 
The former is to rewrite `node.onKey` when `Focus.onKey` is not null, but the latter is unconditional rewriting. 
This test case can reproduce this problem.
```dart
    testWidgets("xxx", (WidgetTester tester) async {
      final GlobalKey key1 = GlobalKey(debugLabel: '1');
      bool? keyEventHandled;
      final FocusNode focusNode = FocusNode(
        onKey: (FocusNode node, RawKeyEvent event) {
          keyEventHandled = true;
          return KeyEventResult.handled;
        }
      );

      await tester.pumpWidget(
        Focus(
          onKey: null,
          focusNode: focusNode,
          child: Container(key: key1),
        ),
      );

      Focus.of(key1.currentContext!).requestFocus();
      await tester.pump();
      await tester.sendKeyEvent(LogicalKeyboardKey.enter);
      expect(keyEventHandled, true);

      keyEventHandled = null;
      // Update the Focus widget.
      await tester.pumpWidget(
        Focus(
          onKey: null,
          focusNode: focusNode,
          child: Container(key: key1),
        ),
      );

      await tester.sendKeyEvent(LogicalKeyboardKey.enter);
      expect(keyEventHandled, true);
    });
```